### PR TITLE
Issue 61 connect sign up modal to backend

### PR DESCRIPTION
**Pull request by jasonkeo**
## Change Summary
-added front end check to signup form
-added sign out to profile
-created create user api

### Change Form
**Fill this up (NA if not available). If a certain criteria is not met, can you please give a reason.**

- [ ] The pull request title has an issue number
- [ ] The change works by "Smoke testing" or quick testing
- [ ] The change has tests
- [ ] The change has documentation

## Other Information
**[Is there anything in particular in the review that I should be aware of?]**

### CODE REVIEW

**Comment by jasonkeo** [comment]
ok idk why but when i went to merge some stuff from main some of the error messages broke so im just gonan leave it like dis

File: client/src/components/ui/SignUpModal.tsx [hunk 1]
```diff
@@ -18,13 +20,133 @@ import {
...
+      firstname: false,
+      lastname: false,
+      phone: false,
+      password: false,
+      confirmpassword: false,
+    };
+
+    e.preventDefault();
+
+    // front end checks for empty field and password
```
Comment (dct0) [inline client/src/components/ui/SignUpModal.tsx:71]: I would move this validation into a separate function to keep things tidy

File: client/src/components/ui/SignUpModal.tsx [hunk 2]
```diff
@@ -18,13 +20,133 @@ import {
...
 function SignUpModal({ isOpen, onClose }: Props) {
+  const { register } = useAuth();
+
+  const [email, setemail] = useState("");
+  const [password, setPassword] = useState("");
+  const [confirmpassword, setconfirmPassword] = useState("");
+  const [firstname, setfirstname] = useState("");
+  const [lastname, setlastname] = useState("");
+  const [city, setCity] = useState("");
+  const [phone, setPhone] = useState("");
```
Comment (dct0) [inline client/src/components/ui/SignUpModal.tsx:41]: Make sure these variables are camelCase

File: client/src/components/ui/SignUpModal.tsx [hunk 3]
```diff
@@ -18,13 +20,133 @@ import {
...
+  const onErrorStyle = "border-2 border-red-500";
+  let [error, setError] = useState({
+    email: false,
+    firstname: false,
+    lastname: false,
+    phone: false,
+    password: false,
+    confirmpassword: false,
+  });
+  let [emsg, setMsg] = useState(Array(5).fill(false));
```
Comment (dct0) [inline client/src/components/ui/SignUpModal.tsx:51]: These should be consts because they don't get reassigned

**Review by dct0 — CHANGES_REQUESTED** [review]
Looks good, some small comments

**Review by dct0 — APPROVED** [review]

**Pull request merged**